### ISSUE TITLE
Vagrant guest cannot create hardlink

### ISSUE THREAD

**Issue by jberger** [TechEmpower/FrameworkBenchmarks#1202]
In building Perl on vagrant, I get strange errors that I have tracked to being unable to create symlinks. In virtualbox the [shared filesystem does not let you do this by default](https://forums.virtualbox.org/viewtopic.php?f=3&t=47014). In order to enable it you have to run 

```
VBoxManage setextradata VM_NAME VBoxInternal2/SharedFoldersEnableSymlinksCreate/SHARE_NAME 1
```

for appropriate values of `VM_NAME` and `SHARE_NAME`. I'm not sure if vagrant can configure these or if it has to be done externally.

**Comment by hamiltont** [comment]
The symlink inside shared folders issue [should be fixed in Vagrant > 1.1] (current is like 1.7 something), so that should not be an issue for anyone. 

According to [the virtualbox issue](https://www.virtualbox.org/ticket/818) they don't support hardlinks in shared folders, and they don't seem to plan on adding that. So, if the perl build currently requires hardlinks, then all I can recommend is that you don't utilize the shared folder. For now, that means commenting out [this line](https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/deployment/vagrant-common/core.rb#L100), and the provision script will [recognize that there is no shared folder and do a git clone instead](https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/deployment/vagrant-common/bootstrap.sh#L112-114)

**Comment by hamiltont** [comment]
Note to self: add vagrant option to disable the mounting, and document using GH_REPO and other variables in bootstrap.sh
